src/frontend/screens/Settings/AppSettings/Languages.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import {ScrollView} from 'react-native';
+import {defineMessages} from 'react-intl';
+import {usePersistedLocale} from '../../../hooks/persistedState/usePersistedLocale';
+import {supportedLanguages} from '../../../lib/intl';
+import {SelectOne} from '../../../sharedComponents/SelectOne';
+import {NativeNavigationComponent} from '../../../sharedTypes/navigation';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.LanguageSettings.title',
+    defaultMessage: 'Language',
+    description: 'Title language settings screen',
+  },
+});
+
+export const Languages: NativeNavigationComponent<'Languages'> = () => {
+  const [locale, setLocale] = usePersistedLocale(store => [
+    store.locale,
+    store.setLocale,
+  ]);
```

Comment:
not entirely sure, but i _think_ returning an array here is not advised by zustand because you lose reference equality in the return value each time the selector is called. the safer option would be to have two invocations instead

```suggestion
  const locale = usePersistedLocale(store => store.locale);
  const setLocale = usePersistedLocale(store => store.setLocale);
```

Reply:
ah yeah here's where i got this from: https://tkdodo.eu/blog/working-with-zustand#prefer-atomic-selectors

alternative is to use specify the `shallow` option, but i think having the two separate invocations works fine